.github/workflows/typespec-validation-all.yaml

```diff
@@ -67,11 +67,11 @@ jobs:
         with:
           # Convert placeholder string 'default' to '', which tells the action to checkout the default ref.
           ref: ${{ matrix.ref == 'next' && 'typespec-next' || '' }}
-          # Checkout all history for all branches.  Required to merge origin/main.
-          fetch-depth: 0
 
-      - name: git merge origin/main
-        run: git -c user.name="azure-sdk" -c user.email="azuresdk@microsoft.com" merge -Xours origin/main
+      - name: git rebase origin/main
+        run: |
+          git fetch origin main --depth=1
```

Comment:
If you had a sparse checkout then I don't even know that you need to do this fetch as it should pull the blobs/commits on-demand as you need them.

Reply:
I assume you are doing the fetch explicitly here because the clone was for the typespec-next branch and you will need to do the fetch of main to do this rebase. I believe this should work and I assume you intentionally made it a rebase instead of a merge.